### PR TITLE
Get rid of getConfig() calls for ManifestJSONParser

### LINKED ISSUE #1925: Get rid of calls of getConfig() in ManifestJSONParser
The constuctor of ManifestJSONParser uses getConfig() to get default argument value for selfHosted and isLanguagePack, so if caller doesn't specify these values, getConfig()  incidentally parses command line arguments, thus if the command does not specify any argument,  yargs output 'Not enough non-option arguments: got 0, need at least 1' error unfortunately.

You can easily confirm the error to run below script on the top of addons-linter source directory.

```
#!/usr/bin/env node

require('./dist/addons-linter')
  .createInstance({
    config: {
      logLevel: 'debug',
      _: ['./tests/fixtures/webextension_scan_file/'],
    },
    runAsBinary: true,
  }).run();
```

### PR DESCRIPTION
**Pull request by hiikezoe**
Fixes #1925.

* [x] The change has been successfully run locally.
* [x] Add tests to cover the changes added in this PR.

### CODE REVIEW

**Comment by hiikezoe** [comment]
gah, cwd option was not supported in node v6?

**Comment by hiikezoe** [comment]
I gave up including the test case.

File: tests/test.yargs-options.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { getDefaultConfigValue } from 'yargs-options';
+
+describe('getDefaultConfigValue()', () => {
+  it('should return the default value', () => {
+    expect(getDefaultConfigValue('self-hosted')).toBeFalsy();
```
Comment (EnTeQuAk) [inline tests/test.yargs-options.js]: Let's change this to use `.toEqual(false)` if we are expecting this or `undefined` etc depending on what value we expect. `falsy` can be many things and explicit is better than implicit.

Comment (hiikezoe) [reply tests/test.yargs-options.js]: Thanks.  That's good to know. I didn't know that what 'toBeFalsy()' exactly does.

**Pull request merged**